sofascoreAcademy/sofascoreAcademy/EventsViewController.swift

```diff
@@ -1,29 +1,114 @@
     private let topBackgroundView = UIView()
     private let menuView = MenuView()
-    private let matchesTableView: UITableView = .init()
     private var eventsViewModel = EventsViewModel()
+    private let eventsHeaderView = EventsHeaderView()
+    private let activityIndicator = UIActivityIndicatorView(style: .medium)
+    private let errorLabel = UILabel()
+    private let matchesTableView: UITableView = .init()
+    private var cancellables = Set<AnyCancellable>()
     
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        eventsHeaderView.delegate = self
         addViews()
         setupConstraints()
         styleViews()
+        activityIndicator.center = view.center
         
+        MenuViewModel.shared.onSportSelectionChanged = { [weak self] selectedSport in
+            self?.handleSportSelectionChanged()
+        }
+        observeEventsViewModel()
+    }
+    
+    private func observeEventsViewModel () {
+        eventsViewModel.$state
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] state in
+                self?.handleState(state)
+            }
+            .store(in: &cancellables)
+    }
+    
+    private func handleState(_ state: EventsViewModel.State) {
+        switch state {
+        case .loading:
+            showLoadingState()
+        case .loaded(_):
+            showLoadedState()
+        case .error:
+            showErrorState()
+        default:
+            break
+        }
+    }
+    
+    func showLoadingState() {
+        activityIndicator.startAnimating()
+        hideError()
+    }
+    
+    func showLoadedState() {
+        activityIndicator.stopAnimating()
+        hideError()
+        matchesTableView.reloadData()
+    }
+    
+    func showErrorState() {
+        activityIndicator.stopAnimating()
+        showError("No data available.")
+    }
+    
+    func showError(_ message: String) {
+        errorLabel.text = message
+        errorLabel.isHidden = false
+    }
+
+    func hideError() {
+        errorLabel.isHidden = true
+    }
+    
+    func handleSportSelectionChanged() {
+        self.eventsViewModel.sections = []
+        self.matchesTableView.reloadData()
+        menuView.updateSelectorPosition(for: MenuViewModel.shared.selectedSport)
         Task {
             await eventsViewModel.loadSections()
-            matchesTableView.reloadData()
         }
     }
     
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        navigationController?.setNavigationBarHidden(true, animated: true)
+        switch eventsViewModel.state {
+        case .idle:
+            Task {
+                await eventsViewModel.loadSections()
+            }
+        default:
+            break
+        }
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        navigationController?.setNavigationBarHidden(false, animated: true)
```

Comment:
Ovo ne moraš raditi, nije te briga što se na drugom screenu dogodi s NavigationBar-om.